### DISCUSSION TITLE
how to compile for windows 32 bits and include sciter.dll

### DISCUSSION THREAD

**Discussion by josej69git** [rustdesk/rustdesk#2216] [Q&A] [answered]
Hi.
I installed and configure all requeriments to make a cargo run / cargo build over Windows and create the rustdesk.exe executable...
i used default = ["use_dasp", "inline"] into [features] tag in Cargo.toml to include all html and other files of src folder.
But i whant to include in the exe file the dll 'sciter.dll' and pack all in only one file. also i want to build a 32 bits version to make it works y all Windows versions...

how to proceed? thanks.

**Comment by ftejada1** [comment]
I have the same issue

**Reply by WeiYea** [reply]
It seems that the source code of "sciter. dll" must be available for static compilation.

I use pyinstaller to package "exe" and "dll" together

**Reply by 291598912** [reply]
Please tell me how to turn off the display of the console when building. I have tried many times, but it is useless. Thank you

**Comment by josej69git** [comment]
i'm new on rust and low knowledge of python, only need compile it with only changing default servers with my own server.
cargo build --release now works in my system, with inline option it includes the ui, but not sciter.dll.
tried:
pyinstaller --onefile --add-binary "sciter.dll;." build.py 
but no result or i can't see results...

**Comment by WeiYea** [comment, marked as answer]
1、build "RustDesk.exe"

2、Write a python code “main.py”
`
from os.path import join
from sys import _MEIPASS
from subprocess import Popen
Popen(join(_MEIPASS, 'RustDesk.exe')).wait()
`

3、Put "main. py", "RustDesk. exe" and "sciter. dll" in the same folder；

4、run `pyinstaller main.py -F -w --add-data "RustDesk.exe;." --add-data "sciter.dll;."`

**Comment by josej69git** [comment]
wow great! thanks!
